### ISSUE TITLE
data2bids crashes with some Elekta data

### ISSUE THREAD

**Issue by dprice80** [fieldtrip/fieldtrip#1113]
Script crashes at line 788 
events_tsv = sortrows(events_tsv, 'onset');
because the events_tsv is not always in the correct format (depending on whether events exist in the data). sortrows reverts to it's non-overloaded method so that the second argument should be either 'ascending' or 'descending'. We have an example dataset we can provide that will replicate the issue. Please contact Rik Henson, CBU, Cambridge for the dataset.

We have collected data from two scanners using the same protocol and the issue only appears in one of those datasets, so we think it must be something specific to the Elekta software version.

**Comment by robertoostenveld** [comment]
Hi @RikHenson, can you shed some light on this? Right now the reported issue is not detailed enough to reproduce (or fix). Would it be possible to share the fif files?

**Comment by dprice80** [comment]
Yes, sorry about that. I thought it would be easier to just reproduce the error yourselves rather than trying to explain it. We don't actually know what the issue is. It's just something about that particular fif file that is causing the error.

**Comment by RikHenson** [comment]
Hi Robert - sorry for the delay - data2bids was actually run by a student of mine, and he says he got the errors below. I have put the corresponding two *.fif files here: ftp://ftp.mrc-cbu.cam.ac.uk/personal/rik.henson/4Robert/.

1. sub-Sub0001_ses-meg1_task-Rest_meg.fif

The error: 
Error using matlab.internal.math.sortrowsParseInputs (line 52)
Sorting direction must be 'ascend' or 'descend'.

Error in sortrows (line 60)
    [col, nanflag, compareflag] = matlab.internal.math.sortrowsParseInputs(A,varargin{:});

Error in data2bids (line 853)
  events_tsv = sortrows(events_tsv, 'onset');

If we change line 853 of data2bids to  “events_tsv = sortrows(events_tsv, 'ascend');” It will work.


2. sub-Sub0018_ses-meg1_task-Rest_proc-tsss.fif

But this does not work by putting “ascend” as an argument.

**Comment by RikHenson** [comment]
@robertoostenveld  see above!

**Comment by robertoostenveld** [comment]
thanks for the files, downloading now. I have more work to do on data2bids today, so I will also have a look at this.

**Comment by robertoostenveld** [comment]
I cannot reproduce the problem, which might be related to a MATLAB version difference. What MATLAB are you using?

Also something else I want to check with you, since the error relates to events: It seems that sub-Sub0001_ses-meg1_task-Rest_meg.fif has zero events, and sub-Sub0018_ses-meg1_task-Rest_proc-tsss.fif has 1 event after about halfway (the file is about 20 minutes long). Is that correct?

**Comment by RikHenson** [comment]
Hi Robert – yes, the events are correct (and unusual) in these files. I am cc’ing Delshad, who originally experienced these errors, and hope he can explain which version of Matlab (and which version of data2bids) he used.

Thanks
Rik

**Comment by schoffelen** [comment]
I am in the process of reviewing some open issues, which are in an unclear, and perhaps unresolvable state.

Is this issue still an issue that can and should be addressed?
